src/app/dataset-view/additional-components/dataset-settings-component/tabs/webhooks/create-subscription-modal/create-subscription-modal.component.html

```diff
@@ -0,0 +1,124 @@
+<div class="d-flex p-4">
+    <div class="flex-grow-1 ms-2">
+        <span class="fs-5 fw-bold">{{
+            subscriptionData ? "Webhook subscription details" : "Create subscription"
+        }}</span>
+    </div>
+    <div>
+        <button
+            type="button"
+            class="btn-close"
+            aria-label="Close"
+            (keydown.esc)="activeModal.close()"
+            (click)="activeModal.close({ action: WebhookSubscriptionModalAction.CLOSE })"
+        ></button>
+    </div>
+</div>
+<div class="px-4">
+    <mat-divider />
+</div>
+
+<div class="body p-4">
+    <form [formGroup]="createSubscriptionForm">
+        <div class="d-flex flex-column">
+            <label for="" class="form-label">Target URL <span class="text-danger">*</span></label>
+            <input type="text" data-test-id="target-url" formControlName="targetUrl" class="w-100 form-control" />
+            <div *ngIf="targetUrlControl?.invalid && (targetUrlControl?.touched || targetUrlControl?.dirty)">
+                <div *ngIf="targetUrlControl?.errors?.required">
+                    <span class="text-danger fs-12" data-test-id="error-name-required"> Target URL is required</span>
+                </div>
+                <div *ngIf="targetUrlControl?.errors?.pattern">
+                    <span class="text-danger fs-12" data-test-id="error-name-pattern">
+                        The value must start with 'https' for security reasons</span
+                    >
+                </div>
+            </div>
+
+            <p class="text-muted fs-12">
+                Must be publicly reachable via HTTPS. Localhost and private Ips are not allowed
+            </p>
+        </div>
+        <div class="d-flex flex-column">
+            <label for="" class="form-label">Subscribed events <span class="text-danger">*</span></label>
+            <ng-select
+                [items]="DROPDOWN_LIST"
+                bindLabel="name"
+                bindValue="value"
+                formControlName="eventTypes"
+                [multiple]="true"
+            >
+                <ng-template ng-option-tmp let-item="item" let-index="index">
+                    <label class="fs-14 inline-block"> {{ item.name }} </label>
+                </ng-template>
+            </ng-select>
+            <div *ngIf="eventTypesControl?.invalid && (eventTypesControl?.touched || eventTypesControl?.dirty)">
+                <div *ngIf="eventTypesControl?.errors?.required">
+                    <span class="text-danger fs-12" data-test-id="error-name-pattern">
+                        Event selection is required</span
+                    >
+                </div>
+            </div>
```

Comment:
I wonder if we could build a directive for all forms that would solve these typical problems, like "required field error", "string length errors", "..".